### DISCUSSION TITLE
Reading current Project file and path

### DISCUSSION THREAD

**Discussion by firelife15** [dmroeder/pylogix#232] [Q&A]
We are building a revision control system. We are wanting ping the current PLC's on the plantfloor and grab the project file and path that resides in the controller properties tab on the PLC.

Does anyone have any idea on reading this particular location?
 
We can get some of the properties, like revision-major/minor, name of the PLC ect.

Thanks for the help!

**Comment by dmroeder** [comment]
I doubt that the PLC has knowledge of the file path.  Major/minor revision is easy, GetDeviceProperties will return the revision.  Getting the controller name will take some additional code, I'll see if I can help with that here in a bit...

**Comment by firelife15** [comment]
Thank you for the quick response. If we could at least get the file name in that spot it would be awesome. 

I am currently online with the plc and will attach screenshot.

When you click on Controller properties.
Select the File tab
The file tab displays the actual current acd file loaded 
in the PLC.
Under that is shows the path to which it was downloaded from.
Next line show time stamps.


![PLC file](https://github.com/dmroeder/pylogix/assets/35383124/5f69bdb9-96dc-44c9-958b-b3b0305bc506)

**Comment by dmroeder** [comment]
One attempt to find out for sure if that is actually stored in the PLC would be to run wireshark, then open controller properties, navigate to that tab.  See if you see that data come across in wireshark.  I doubt it is though, just because you see that there doesn't mean that it is actually being pulled from the controller.

I think the controller is unaware of all of that information, that is being read from from the file and displayed.  The time stamps are likely from the file metadata.  Make changes while offline, that data changes along with it.

When I said I might be able to get the name, I meant the controller name, not the file name.  In your case: FL000510_72009TX

**Comment by firelife15** [comment]
Yes I have already been able to get the Controller name.

I will open wireshark and see if anything shows up when I do that.

**Comment by TheFern2** [comment]
I would probably go about a different direction for this versioning system. In most cases plants have engineering workstations, for this project I would probably setup one that no one can touch. Keep all plc acd's in this one pc, could make one folder for each PLC, etc, every week or so, use autoit to do automation so that rslogix/studio5k tries to go online, and if it can't then something has changed, then perform an upload.

With both old backup and new upload, convert them to l5x, then you can easily make comparisons. Trying to fetch a file name is probably going to be a dead end.

Though, if you have constant uncontrolled changes, I would say there is a break in the process already that needs fixing. Engineering changes should have a request, approval, backup before change, put backup in some safe location, test procedure, etc. I am sure all that's being done.

**Reply by firelife15** [reply]
I totally agree our system right now is broke. We are just starting to unfold 20+ years of to many hands in the programs. Currently right now just trying to get all the programs and systems categorized and verified to what we have on the floor.
